Detectors/TPC/reconstruction/include/TPCReconstruction/KrBoxClusterFinder.h

```diff
@@ -0,0 +1,166 @@
+/// localMaxima() yields a vector of a tuple with three int
+/// Loop over this vector and give the tuple to buildCluster()
+/// Write to tree or something similar
+
+#ifndef ALICEO2_TPC_KrBoxClusterFinder_H_
+#define ALICEO2_TPC_KrBoxClusterFinder_H_
+
+//#pragma link C++ class std::vector < KrCluster> + ;
+
+#include "TCanvas.h"
+#include "TColor.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TTree.h"
+
+#include "TPCBase/Digit.h"
+#include "TPCReconstruction/KrCluster.h"
+
+#include <array>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+namespace o2
+{
+namespace tpc
+{
+//
+// struct cluster {
+// public:
+//   float totCharge = 0;
+//   float maxCharge = 0;
+//   int size = 0;
+//   // means and sigmas are weighted averages/sigmas
+//   float meanPad = 0;
+//   float meanRow = 0;
+//   float meanTime = 0;
+//   float sigmaPad = 0;
+//   float sigmaRow = 0;
+//   float sigmaTime = 0;
+//   int sector = 0;
+// };
+
+/// KrBoxClusterFinder class can be used to analyze X-ray and Krypton data
+///
+/// It creates a 3D map of all digits in one event. Afterwards the Function
+/// findLocalMaxima() can be used to get the coordinates of all local maxima.
+/// Around these maxima, the cluster is then built with the function
+/// buildCluster. In a box around the center (defined by maxClusterSize....) it
+/// looks for digits that can be assigned to the cluster.
+class KrBoxClusterFinder
+{
+ public:
+  /// Constructor:
+  explicit KrBoxClusterFinder(std::vector<o2::tpc::Digit>* eventSector); // Creates 3D Map
+
+  /// After the map is created, we look for local maxima with this function:
+  std::vector<std::tuple<int, int, int>> findLocalMaxima();
+
+  /// After finding the local maxima, we can now build clusters around it.
+  /// It works according to the explanation at the beginning of the header file.
+  KrCluster buildCluster(int clusterCenterPad, int iRow, int iTime);
+
+ private:
+  // These variables can be varied
+  int mMaxClusterSizeTime = 2; ///< The "radius" of a cluster in time direction
+  int mMaxClusterSizePad = 2;  ///< radius in pad direction
+
+  int mMaxClusterSizeRow = 2; ///< radius in row direction
+
+  // Todo: Differentiate between different ROCS:
+  // int mMaxClusterSizeRowIROC = 3;  // radius in row direction
+  // int mMaxClusterSizeRowOROC1 = 2; // radius in row direction
+  // int mMaxClusterSizeRowOROC2 = 2; // radius in row direction
+  // int mMaxClusterSizeRowOROC3 = 1; // radius in row direction
+
+  int mQThresholdMax = 10;        ///< the Maximum charge in a cluster must exceed this value or it is discarded
+  int mQThreshold = 1;            ///< every charge which is added to a cluster must exceed this value or it is discarded
+  int mMinNumberOfNeighbours = 1; ///< amount of direct neighbours required for a cluster maximum
+
+  /// Here the map is defined where all digits are temporarily stored
+  std::vector<std::vector<std::vector<int>>> mMapOfAllDigits;

```

Comment:
Please put the vector after all trivial data types, so after `mTempCluster`

Reply:
Done.